project/src/main/java/com/google/sps/utilities/SpannerTasks.java

```diff
@@ -600,4 +606,84 @@ public Void run(TransactionContext transaction) throws Exception {
     }
     return results;
   }
+ 
+  /**
+   * Add event results to persistent storage index by inserting the keyword into the keywords table
+   * if it dose not exist and adding a corresponding row in the results table.
+   *
+   * @param results the event results to add to the index
+   */
+  public static void addResultsToPersistentStorageIndex(List<EventResult> results) {
+    for (EventResult result : results) {
+      SpannerClient.getDatabaseClient()
+          .readWriteTransaction()
+          .run(
+              new TransactionCallable<Void>() {
+                @Override
+                public Void run(TransactionContext transaction) throws Exception {
+                  String eventId = result.getEventId();
+                  String keyword = result.getKeyword();
+                  float ranking = result.getRanking();
+                  Struct row =
+                      transaction.readRow(
+                          KEYWORDS_TABLE, Key.of(keyword), Collections.singleton(NAME));
+                  String keywordId = row == null ? UUID.randomUUID().toString() : row.getString(0);
+                  transaction.buffer(
+                      Mutation.newInsertBuilder(KEYWORDS_TABLE)
+                          .set(KEYWORD_ID)
+                          .to(keywordId)
+                          .set(NAME)
+                          .to(keyword)
+                          .build());
+                  transaction.buffer(
+                      Mutation.newInsertBuilder(RESULTS_TABLE)
+                          .set(KEYWORD_ID)
+                          .to(keywordId)
+                          .set(EVENT_ID)
+                          .to(eventId)
+                          .set(RANKING)
+                          .to(ranking)
+                          .build());
+                  return null;
+                }
+            });
+    }
```

Comment:
nit: consistent spacing? seems to be a combo of 2 or 4 spaces

Reply:
Copy pasted this from linter.